test/system/user_test.rb

```diff
@@ -0,0 +1,97 @@
+require 'application_system_test_case'
+
+class UsersTest < ApplicationSystemTestCase
+  #########################
+  # Tests for user index
+  #########################
+
+  test 'displays the username of all users' do
+    visit users_path
+    sleep(1)
+
+    assert_text 'username1'
+    assert_text 'username2'
+  end
+
+  test 'displays the profile picture for each user' do
+    visit users_path
+
+    assert_selector('img[alt="Profile photo"]', count: 2)
+  end
+
+  test 'displays the number of posts each user has written' do
+    visit users_path
+
+    assert_text "posts counter: #{users(:one).posts_counter}"
+  end
+
+  test 'redirects to the user show page' do
+    visit users_path
+    click_link users(:one).name
+
+    assert_current_path(user_path(users(:one)))
+  end
```

Comment:
- [ ] As required for the project, kindly ensure that you have a specific `Spec_file` which only tests `User` index page as right now you put all the user tests in the same file